smile: init at 2.8.1

foo-dogsquared opened the pull request:
###### Description of changes

Add [Smile](https://smile.mijorus.it/), an emoji picker for Linux.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

foo-dogsquared committed:
smile: init at 2.8.1

foo-dogsquared commented:
Updated the package to 2.3.1. Also updated the package file a bit. This should also fix the error from ofborg build.

oluceps commented:
Run result on gnome wayland.
```
> ./smile 

(process:4127576): Gdk-WARNING **: 00:40:06.509: No such backend: *
Traceback (most recent call last):
  File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/bin/.smile-wrapped", line 45, in <module>
    from smile import main
  File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/main.py", line 9, in <module>
    from .Picker import Picker
  File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/Picker.py", line 24, in <module>
    from .components.CustomTagEntry import CustomTagEntry
  File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/components/CustomTagEntry.py", line 4, in <module>
    from .CustomPopover import CustomPopover
  File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/components/CustomPopover.py", line 5, in <module>
    gi.require_version('Adw', '1')
  File "/nix/store/bybx4aya78qc5bw8yb94bsiayh3afmxx-python3.10-pygobject-3.44.1/lib/python3.10/site-packages/gi/__init__.py", line 126, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace Adw not available
```

As the last review shows, missing dependencies. Add `libadwaita` into `buildInputs` could solve this.

foo-dogsquared commented:
> Run result on gnome wayland.
> 
> ```
> > ./smile 
> 
> (process:4127576): Gdk-WARNING **: 00:40:06.509: No such backend: *
> Traceback (most recent call last):
>   File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/bin/.smile-wrapped", line 45, in <module>
>     from smile import main
>   File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/main.py", line 9, in <module>
>     from .Picker import Picker
>   File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/Picker.py", line 24, in <module>
>     from .components.CustomTagEntry import CustomTagEntry
>   File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/components/CustomTagEntry.py", line 4, in <module>
>     from .CustomPopover import CustomPopover
>   File "/nix/store/lv38n9bb0s5n2h2jvwv90f714ic52a39-smile-2.3.1/share/smile/smile/components/CustomPopover.py", line 5, in <module>
>     gi.require_version('Adw', '1')
>   File "/nix/store/bybx4aya78qc5bw8yb94bsiayh3afmxx-python3.10-pygobject-3.44.1/lib/python3.10/site-packages/gi/__init__.py", line 126, in require_version
>     raise ValueError('Namespace %s not available' % namespace)
> ValueError: Namespace Adw not available
> ```
> 
> As the last review shows, missing dependencies. Add `libadwaita` into `buildInputs` could solve this.

Thanks for looking out. I just rebased this to recent revisions from master branch and updated it to 2.8.1. Please have a look again.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1176

pbsds commented on pkgs/by-name/sm/smile/package.nix:
```diff
@@ -0,0 +1,77 @@
+{ stdenv
+, lib
```
```suggestion
{ lib
```

pbsds reviewed (changes requested):
Runs fine

Result of `nixpkgs-review pr 235355` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>smile</li>
  </ul>
</details>

The file `/share/smile/smile/smile.in` is executable and has the shebang `#!@PYTHON@`. One of those is likely not intentional.

Aleksanaa commented:
Done in https://github.com/NixOS/nixpkgs/pull/284708
